### DISCUSSION TITLE
Gitlab storage not working with wiki.js

### DISCUSSION THREAD

**Discussion by Mylenu** [requarks/wiki#6410] [Help / Questions]
Hi,

I installed wiki js like the official documentation : https://docs.requarks.io/install/ubuntu
I generated a public key and a private key on my wiki server (test server).
The gitlab is another server (production server).

I wanted to have the doc of my gitlab but it doesnt work.

Settings :
Authentification type : SSH
Repository : git@git.mydomain/docs.git
Branch : master
SSH private key mode : contents
SSH private key contents : my private key
Verify SSL Cert : ON

I have an unexpected error. I tried to deploy a token on my projet with my public key from wiki.js and grant him write permission but it doesnt do anything.
I tried too set a SSH Keys into my profile with usage type "Authentification & Signing " but doesnt work either.

I know that the official documentation is only for github and gitlab is comming soon but i wanted to know if it is possible to set it or the options actually available is no good. Or it's just that i dont know how to configure it. (Im a newbie in those thing)

I know that the configuration with github like the official documentation is working well.

I would like this to work and i can provide more information if necessary.

**Comment by Mylenu** [comment]
I have figured out something, it could be that the ssh isnt allowed on the server Gitlab so it's normal that it doesnt work.
I configured wiki.js with the basic type and i created a user "wiki" on my gitlab to access the data

**Comment by Doneven2012** [comment]
Hey, I don't know if you got this to work, but maybe I can help. I'd say to create a new SSH key with the command in the documentation, I use Windows Subsystem for Linux to create one, but you can use your favorite tool. "ssh-keygen -t rsa -b 4096"

The private key will go on WikiJS, like you were doing it, with the key mode to "Contents" and the content pasted there. Make sure to include the lines -----BEGIN OPENSSH PRIVATE KEY----- and -----END OPENSSH PRIVATE KEY-----.

The public key should go here in GitLab: 
-  Open your project's page
-  Click Settings on the left followed by Repository<
-  Scroll down to "Deploy keys"
-  Paste your public key and make sure to tick  "Grant write permissions to this key"

My main branch on Gitlab was "main" instead of "master", maybe that's part of your issue? You didn't provide an error message so hard to say (there should be one on the "Storage" page where you configure Gitlab, under "Status")

Another issue I found was with "Repository URI", use the entire URI you obtain when cloning the repository. "ssh://git@git.mydomain/docs.git". The WikiJS git config page makes it seem like you have to remove ssh://, but when I did it simply did not work. 

Finally, if you do get to the point where Git works (Green checkmark near "Git" on WikiJS under "Status", which is under the list of storage options), you can click "Add untracked changed", wait a couple minutes then force a sync..


Hopefully that helps, I know it's a lot of info and I wasn't sure where to start or where to end! But that's what made it work for me.

**Reply by ahintit** [reply]
Thank you) Your comment helped me a lot)
My steps

Gitlab
1. Grant write permissions to this key - for sync repo

Wikijs:
1. Full address to ssh with "ssh://" part
2. Included "-----BEGIN OPENSSH PRIVATE KEY----- and -----END OPENSSH PRIVATE KEY-----" into private part of the key

Now it works))
